validmind/vm_models/dataset.py

```diff
@@ -479,7 +603,46 @@ def assign_predictions(  # noqa: C901 - we need to simplify this method
             )
 
             prediction_values = np.array(model.predict(x_only))
-            self.__assign_prediction_values(model, pred_column, prediction_values)
+
+            # Check if the prediction values are probabilities
+            if _is_probability(prediction_values):
+
+                threshold = 0.5
```

Comment:
Why are we hard coding threshold here? It would be model developer's job to convert probabilities into `y_true`.

Reply:
I think in this case it's fine since the developer only passed a model and is letting us decide the threshold.

I would suggest we add a warning here that tells the user something like "using default threshold of 0.5 to assign class labels".

If the user has their own threshold they will need to pass prediction_values and prediction_probabilities manually.

Reply:
Nevermind, I see that you already do this here: https://github.com/validmind/developer-framework/pull/43/files#diff-19f766812f0e37e809d1eda669f0a45160d04616ba506156606dc1965cbcbce5R618